DOCS-7450-reconnections-duke (#1289)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/reconnection-strategy-about.adoc

endif::[]
:keywords: anypoint, studio, reconnection strategies, reconnection strategy, retry policies, retry

> When an operation in a Mule application fails to connect to an external server, the default is for the operation to fail immediately due to the connectivity issue. Upon failing, the operation returns a connectivity error. For example, if an operation in the FTP connector fails to connect to an FTP server, the default is to throw an `FTP:CONNECTIVITY` error. The error name prepends a namespace that is specific to the connector, such as `FTP:CONNECTIVITY`, `HTTP:CONNECTIVITY`, or `DB:CONNECTIVITY` (for the Database connector).

Reviewer: ```suggestion
When an operation in a Mule application fails to connect to an external server, the default is for the operation to fail immediately. Upon failing, the operation returns a connectivity error. For example, if an operation in Anypoint Connector for FTP (FTP Connector) fails to connect to an FTP server, the default is to throw an `FTP:CONNECTIVITY` error. The error name prepends a namespace that is specific to the connector, such as `FTP:CONNECTIVITY`, `HTTP:CONNECTIVITY`, or `DB:CONNECTIVITY` (for Anypoint Connector for Database (Database Connector).
```

You can modify this behavior by setting a reconnection strategy on the operation:

> * None: Default behavior, which immediately returns a connectivity error if the attempt to connect is unsuccessful. 

Reviewer: GLOBALLY follow style guidelines for bulleted lists:
```suggestion
* None
+
+ Default behavior, which immediately returns a connectivity error if the attempt to connect is unsuccessful. 
```

> * Standard (`reconnect`): Sets a number of reconnection attempts and the interval at which to execute them before returning a connectivity error. For example, you might set a `count` of `5` reconnection attempts at a `frequency` of every `4000` milliseconds (ms). Defaults to `2` reconnection attempts at a frequency of `2000` ms.

Reviewer: ```suggestion
* Standard (`reconnect`)
+
+ Sets a number of reconnection attempts and the interval at which to execute them before returning a connectivity error. For example, you might set a `count` of `5` reconnection attempts at a `frequency` of every `4000` milliseconds (ms). Defaults to `2` reconnection attempts at a frequency of `2000` ms.
```

</flow>
----
+

> * Forever (`reconnect-forever`): Attempts to reconnect at a given interval, for example, every `4000` ms for as long as the application runs. Defaults to every `2000` ms. A failure to connect never produces a connectivity error.

Reviewer: ```suggestion
* Forever (`reconnect-forever`)
+
+ Attempts to reconnect at a given interval, for example, every `4000` ms for as long as the application runs. Defaults to every `2000` ms. A failure to connect never produces a connectivity error.
```
